src/exchanges.js

```diff
@@ -35,10 +21,16 @@ var Publisher = function(conn, channel, entries, exchangePrefix, options) {
   this._channel = channel;
   this._entries = entries;
   this._options = options;
-  if (options.drain) {
-    assert(options.component, "component name for statistics is required");
-    assert(options.process,   "process name for statistics is required");
-    this._reporter = ExchangeReports.reporter(options.drain);
+  if (options.drain || options.component) {
+    console.log('taskcluster-lib-stats is now deprecated!\n' +
+                'Use the `monitor` option rather than `drain`.\n' +
+                '`monitor` should be an instance of taskcluster-lib-monitor.\n' +
+                '`component` is no longer needed. Prefix your `monitor` before use.');
+  }
+
+  var monitor = null;
+  if (options.monitor) {
+    monitor = options.monitor.prefix('pulse-publisher');
```

Comment:
Hmm... I have mixed feelings about this... 
Doing all prefixing inside api/pulse-publisher is simple, but on the other hand it becomes rather awkward if you ever need two instances of API or pulse-publisher in the same component...

I guess in such case we just prefix them `v1` and `v2` and then it becomes `v1.api.<whatever>` and `v2.api.<whatever>`...

@imbstack, yes, let's just do what you do now... But let's change the prefix to `exchanges` or something simpler..